### PR TITLE
[docs] Add docs about span macros

### PR DESCRIPTION
**Pull request by smyrick**
The docs reference the macros available but only mention how to use the metric ones when we do support the span macros too

### CODE REVIEW

File: docs/source/customizations/native.mdx [hunk 1]
```diff
@@ -281,6 +281,22 @@ tracing::info!(
...
+Make sure to [enable OpenTelemtry tracing](../configuration/telemetry/exporters/tracing/overview) in your configuration if you want to custom traces generated and linked by the Router.
+
+</Note>
+
+To create custom spans and traces you can use the [tracing macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate a span.
+
+```rust
+use tracing::{Level, span};
+
+span!(target: "app_spans", Level::TRACE, "my span");
```
Comment (BrynCooke) [inline docs/source/customizations/native.mdx:296]: It's probably best to use info_span! as the level directly corresponds to the log level which most people will set to info.

**Review by BrynCooke — APPROVED** [review]

File: docs/source/customizations/native.mdx [hunk 2]
```diff
@@ -234,17 +234,17 @@ plugins:
     # Any values here are passed to the plugin as part of your configuration
 ```
 
-## Add custom metrics
+## Using macros
+To create custom metrics, traces, and spans you can use the [tracing macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate events and logs.
```
Comment (shorgi) [inline docs/source/customizations/native.mdx:238]:
```suggestion
To create custom metrics, traces, and spans, you can use [`tracing` macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate events and logs.
```

File: docs/source/customizations/native.mdx [hunk 3]
```diff
@@ -234,17 +234,17 @@ plugins:
...
 <Note>
 
 Make sure to [enable prometheus metrics](../configuration/telemetry/exporters/metrics/prometheus) in your configuration if you want to have metrics generated by the Router.
 
 </Note>
 
-### Using macros
-
-To create your custom metrics in [Prometheus](https://prometheus.io/) you can use the [tracing macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate an event.
+To create your custom metrics in [Prometheus](https://prometheus.io/) you can use the [event macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate an event.
```
Comment (shorgi) [inline docs/source/customizations/native.mdx:247]:
```suggestion
To create your custom metrics in [Prometheus](https://prometheus.io/) you can use [event macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate an event.
```

File: docs/source/customizations/native.mdx [hunk 4]
```diff
@@ -281,6 +281,22 @@ tracing::info!(
...
 ```
 
+### Add custom spans
+<Note>
+
+Make sure to [enable OpenTelemtry tracing](../configuration/telemetry/exporters/tracing/overview) in your configuration if you want to custom traces generated and linked by the Router.
+
+</Note>
+
+To create custom spans and traces you can use the [tracing macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate a span.
```
Comment (shorgi) [inline docs/source/customizations/native.mdx:291]:
```suggestion
To create custom spans and traces you can use [`tracing` macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate a span.
```

File: docs/source/customizations/native.mdx [hunk 5]
```diff
@@ -281,6 +281,22 @@ tracing::info!(
 );
 ```
 
+### Add custom spans
+<Note>
+
+Make sure to [enable OpenTelemtry tracing](../configuration/telemetry/exporters/tracing/overview) in your configuration if you want to custom traces generated and linked by the Router.
```
Comment (shorgi) [inline docs/source/customizations/native.mdx:287]:
```suggestion
Make sure to [enable OpenTelemetry tracing](../configuration/telemetry/exporters/tracing/overview) in your configuration if you want customize the traces generated and linked by the router.
```

**Review by shorgi — CHANGES_REQUESTED** [review]

**Comment by abernix** [comment]
@shorgi Could you re-review? Thanks!

**Review by shorgi — APPROVED** [review]

**Pull request merged**